contents/ruby/Rails.md

```diff
@@ -0,0 +1,170 @@
+<frontmatter>
+  title: Introduction to Ruby on Rails
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Ruby on Rails
+
+Authors: [Chattoraj Ayush](https://github.com/AyushChatto)
+
+## Ruby on Rails overview
+
+**Ruby on Rails** is a web application framework written in the Ruby programming language. It was designed with the intention of making programming web applications easier and quicker by reducing the amount of code you write. Rails achieves this by being an "opinionated framework", which means that the developers believe that there is a certain approach - often called "The Rails Way" - that is ideal for building most types of web applications. By adhering to this approach, Rails allows you to focus primarily on the logic, and fills in much of the boilerplate code for you. 
+
+The Rails philosophy includes two major principles, namely:
+1. **Don't Repeat Yourself (DRY)**
+2. **Convention over Configuration** 
+
+### The Rails Way 
+
+One of the core tenets of Rails is "Convention over Configuration", which means that many decisions regarding the structure of your Rails application are made for you in advance, so that there is less effort needed to come up with your own set of best practices. This is perhaps most evident in Rails' decision to follow a Model-View-Controller (MVC) architecture by default.
+
+The structure of an app written in the MVC framework is as follows:
```

Comment:
Replace this with `Here's how a Rails application is structured:`

Because the filenames etc. in the diagram are specific to Rails. Also, you can't really... write an app in the MVC framework. The Rails framework does _follow an MVC architecture_, though.

Reply:
Whoops meant architecture here. Fixed it :)